src/waitlist.cpp

```diff
@@ -95,36 +134,25 @@ bool WaitingList::clientLogin(const Player* player)
 		return false;
 	}
 
-	slot = priorityWaitList.size();
+	slot = info->priorityWaitList.size();
 	if (player->isPremium()) {
-		priorityWaitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
+		info->priorityWaitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
 	} else {
-		slot += waitList.size();
-		waitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
+		slot += info->waitList.size();
```

Comment:
Hey, I think this is wrong. Why is the free account player slot added the size of the free waitlist? Shouldn't it be added the size of the PREMIUM waitlist, i.e. the first free player slot is immediately after the last premium player?